### ISSUE TITLE
How to execute modules requiring 'document' in plugin code?

### ISSUE THREAD

**Issue by eyalroth** [gatsbyjs/gatsby#14793]
## Summary
I'm trying to execute a third-party module which uses `document` (I'm assuming it modifies the HTML). I can execute the module in the `render()` method of my pages' component, but I'm actually looking to write this functionality as a plugin; meaning, I want to avoid invoking the module inside the react component of my pages / templates and execute it after the pages are created.

## In more detail
I'm trying to write a plugin which transforms markdown footnotes into popovers / tooltips via the module [tippy.js](https://github.com/atomiks/tippyjs) package, which also has a [React wrapper](https://github.com/atomiks/tippy.js-react).

My tactic is first to transform the footnote nodes in the markdown AST to HTML which the tippy module can later modify (by searching for HTML elements with a custom class). I succeeded in transforming the AST into the relevant HTML by implementing a local remark plugin. I was then able to let tippy do its thing in the `render()` function of my pages' template.

That worked, but I want to execute the tippy module as part of the plugin as well, and I'm unsure on how to do it, as the module requires a global `document`. I tried implementing all kind of API calls in the plugin's `gatsby-node.js`, `gatsby-browser.js`, `gatsby-ssr.js` files, but with no success.

Help would be appreciated :)

**Comment by eyalroth** [comment]
Well, it appears all I needed was 15 more minutes of research or so. Turns out I can execute the module from within the `onRouteUpdate` function in the plugin's `gatsby-browser.js`. The catch is I needed to `import` the module and not `require` it (otherwise I get the "not a function" exception). Something like this:

```javascript
// plugins/my-plugin/gatsby-browser.js
import tippy from 'tippy.js'

export function onRouteUpdate() {
    tippy(".custom-class-name")
}
```

I'm not entirely sure this is the right way to achieve this, but I gathered this should be acceptable given that a very similar library named [littlefoot](https://github.com/goblindegook/littlefoot) -- which transforms footnotes into something prettier -- has these exact instructions for Gatsby users.
